app/tests/components_tests/test_partial_templates.py

```diff
@@ -0,0 +1,121 @@
+)
+@pytest.mark.parametrize(
+    "component_interface, component_interface_value,expected_snippet",
+    (
+        (
+            dict(  # VALUE
+                kind=InterfaceKindChoices.FLOAT,
+            ),
+            dict(
+                value=42,
+            ),
+            "<pre",
+        ),
+        (  # VALUE CHART
+            dict(
+                kind=InterfaceKindChoices.CHART,
+            ),
+            dict(
+                value={
+                    "$schema": "https://vega.github.io/schema/vega-lite/v5.json",
+                    "description": "A simple bar chart with embedded data.",
+                    "data": {
+                        "values": [
+                            {"a": "A", "b": 28},
+                            {"a": "B", "b": 55},
+                        ]
+                    },
+                    "mark": "bar",
+                    "encoding": {
+                        "x": {
+                            "field": "a",
+                            "type": "nominal",
+                            "axis": {"labelAngle": 0},
+                        },
+                        "y": {"field": "b", "type": "quantitative"},
+                    },
+                },
+            ),
+            "vega-lite-chart",
+        ),
+        (
+            dict(  # FILE
+                kind=InterfaceKindChoices.PDF,
+                store_in_database=False,
+            ),
+            dict(),
+            "<a ",
+        ),
+        (  # FILE Thumbnail
+            dict(
+                kind=InterfaceKindChoices.THUMBNAIL_JPG,
+                store_in_database=False,
+            ),
+            dict(),
+            "<img",
+        ),
+        (  # IMAGE
+            dict(
+                kind=InterfaceKindChoices.IMAGE,
+                store_in_database=False,
+            ),
+            dict(),
+            "<a ",
+        ),
+        (  # Broken / fallback
+            dict(
+                kind=InterfaceKindChoices.SEGMENTATION,
+                store_in_database=False,
+            ),
+            dict(),
+            "cannot be displayed",
+        ),
+    ),
+)
+@pytest.mark.django_db
+def test_civ(
+    template, component_interface, component_interface_value, expected_snippet
+):
+    ci = ComponentInterfaceFactory(**component_interface)
+    civ = ComponentInterfaceValueFactory.build(
+        interface=ci, **component_interface_value
+    )
+
+    if ci.kind == InterfaceKindChoices.PDF:
+        civ.file.save("file.pdf", ContentFile(b"%PDF-1.4\n"))
+
+    if ci.kind == InterfaceKindChoices.THUMBNAIL_JPG:
+        civ.file.save(
+            "thumbnail.jpg",
+            ContentFile(b"<bh:ff><bh:d8><bh:ff><bh:e0><bh:00><bh:10>JFIF"),
+        )
+
+    if ci.kind == InterfaceKindChoices.IMAGE:
+        civ.image = ImageFactory()
+        ImageFileFactory(image=civ.image)
+
+    # Actually create the CIV
+    if ci.kind != InterfaceKindChoices.SEGMENTATION:  # Intentionally broken
```

Comment:
Just for my information, why?

Reply:
To test the fallback which represents a graceful fail route. This stems from that we create the CIV generally by first binding the interface and later bind the image/value/file. I am not 100% convinced we do or if it's even possible to do these atomic.

Perhaps 'incomplete' is a better term.